include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -232,11 +232,13 @@ ORT_API_STATUS(OrtSetSessionLogId, _Inout_ OrtSessionOptions* options, const cha
 ORT_API_STATUS(OrtSetSessionLogVerbosityLevel, _Inout_ OrtSessionOptions* options, int session_log_verbosity_level);
 
 // Set Graph optimization level.
-// Available options are : 0, 1, 2.
-// 0 -> Disable all optimizations
-// 1 -> Enable basic optimizations
-// 2 -> Enable all optimizations
-ORT_API_STATUS(OrtSetSessionGraphOptimizationLevel, _Inout_ OrtSessionOptions* options, int graph_optimization_level);
+typedef enum GraphOptimizationLevel {
+  ORT_DISABLE_ALL,
+  ORT_ENABLE_BASIC,
+  ORT_ENABLE_ALL
```

Comment:
Maybe set explicit values here. ORT_ENABLE_ALL could be a very large value so you can add things between basic and all.

Reply:
I did set ORT_ENABLE_ALL to a large value initially, but then reverted it because the Python API doesn't use this enum and hence it would create inconsistency in the value of ORT_ENABLE_ALL 
 between different language APIs.

Reply:
The enum proposed in the email thread --

DISABLE_ALL = TransformerLevel::Default
ENABLE_BASIC = TransformerLevel::Level1
ENABLE_ADVANCED (or ENABLE_EXTENDED) = TransformerLevel::Level3
ENABLE_ALL = TransformerLevel::Level2

Why ENABLE_ALL = level2 and ENABLE_EXTENDED = level3?

if higher-valued enum always includes all optimizations of the lower values, we can be more specific in ENABLE_ADVANCED, e.g. ENABLE_LAYOUT.

Reply:
Fixed this.